MagneticField/Engine/python/volumeBasedMagneticField_dd4hep_160812_cfi.py

```diff
@@ -0,0 +1,95 @@
+import FWCore.ParameterSet.Config as cms
+
+
+# This cfi contains everything needed to use the VolumeBased magnetic
+# field engine version 160812 built using dd4hep for the geometry.
+#
+# PLEASE DO NOT USE THIS DIRECTLY
+# Always use the standard sequence Configuration.StandardSequences.MagneticField_cff
+
+
+DDDetectorESProducer = cms.ESSource("DDDetectorESProducer",
```

Comment:
Good point, will do

Reply:
Funny, enough it does not work if I rename it to eg 

> 
process.DDDetectorMFESProducer = cms.ESSource("DDDetectorESProducer", ...

I then get:

> An exception of category 'NoProxyException' occurred while
>    [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
>    [1] Running path 'p'
>    [2] Calling method for module testMagGeometryAnalyzer/'test'
>    [3] Using EventSetup component DD4hep_VolumeBasedMagneticFieldESProducer/'MagneticFieldESProducer' to make data MagneticField/'' in record Id
>    [4] Using EventSetup component DDCompactViewMFESProducer/'' to make data DDCompactView/'magfield' in record IdealMagneticFieldRecord
> Exception Message:
> No data of type "DDDetector" with label "magfield" in record "IdealMagneticFieldRecord"
> 

It can be quickly reproduced editing https://github.com/cms-sw/cmssw/blob/5d038bc896ae144f071cd0ab83e07c36b3970757/MagneticField/GeomBuilder/test/python/testMagGeometry.py#L40 .

@cvuosalo, @ianna, any idea of why this happens and how to avoid this?

@slava77 I will go on with the other  fixes (in a new PR) while this is getting understood.

Reply:
I'm confused in what's failing and what is edited.
volumeBasedMagneticField_dd4hep_160812_cfi.py is currently included only in regression.py, but it runs only testMagneticField, while the error message is coming from testMagGeometryAnalyzer.

Reply:
I am trying to apply the modification  in cmssw/MagneticField/GeomBuilder/test/python/testMagGeometry.py first, as mentioned in my message above. This contains a (stripped-down) copy of the same config, for the purpose of testing the geometry. 
Of course I need to fix volumeBasedMagneticField_dd4hep_160812_cfi.py as well, but the problem would be the same.